builtin/commit-graph.c

```diff
@@ -9,7 +9,7 @@
 
 static char const * const builtin_commit_graph_usage[] = {
 	N_("git commit-graph verify [--object-dir <objdir>] [--shallow] [--[no-]progress]"),
-	N_("git commit-graph write [--object-dir <objdir>] [--append|--split] [--reachable|--stdin-packs|--stdin-commits] [--[no-]progress] <split options>"),
+	N_("git commit-graph write [--object-dir <objdir>] [--append|--split] [--reachable|--stdin-packs|--stdin-commits] [--bloom] [--[no-]progress] <split options>"),
```

Comment:
Add a `TODO` to document this change in `Documentation/git-commit-graph.txt`. You may need to add a description of them in `Documentation/technical/commit-graph.txt` and link to that.

Reply:
@garimasi514 you are probably going to hate me for this late request, but the recent paper about "xor filters better than bloom filters" made me realize something: this option should be about _what_ it does, not _how_ it does it.

It doesn't matter to the user that we are using Bloom filters to (probabilisticly) store changed paths. It matters that we are storing some information on their changed paths and will speed up their log queries.

For now, add to your commit message a request for suggestions to name this option. Suggest things like `--paths` or `--changed-paths`.

Even the file format can become "Bloom agnostic" by using the first byte of the data chunk as a "version" number. It doesn't need to be the hash _function_ version, but instead the data storage version. A version 2 could use a different storage mechanism entirely.

(Making a note for later: we will want to be sure that we only warn on an unexpected version number then continue without the data. We've been bitten by failing on an unexpected version, as that is bad for the compatibility story.)

Aside: the XOR filters are not useful for our case, since you cannot re-use the hash of a key between two filters.

Reply:
@dstolee I will rename --bloom to "--changed-paths" and put in an RFC note. And I don't hate you one bit. I read the article about that paper today morning too after you liked it on twitter :)